packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -70,9 +70,46 @@ export const checkIfIsAvailable = ({
 };
 
 export async function getEventType(input: TGetScheduleInputSchema) {
+  const { eventTypeSlug, usernameList } = input;
+  let eventTypeId = input.eventTypeId;
+
+  if (eventTypeId === undefined && eventTypeSlug && usernameList && usernameList.length === 1) {
+    // If we only have the slug and usernameList, we need to get the id first
+    const username = usernameList[0];
+    const userId = await getUserIdFromUsername(username);
+    let teamId;
+
+    if (!userId) {
+      teamId = await getTeamIdFromSlug(username);
+      if (!teamId) {
+        throw new TRPCError({
+          message: "User or team not found",
+          code: "NOT_FOUND",
+        });
+      }
+    }
+
+    const eventType = await prisma.eventType.findFirst({
+      where: {
+        slug: eventTypeSlug,
+        ...(teamId ? { teamId } : {}),
+        ...(userId ? { userId } : {}),
+      },
+      select: {
+        id: true,
+      },
+    });
+
+    if (!eventType) {
+      throw new TRPCError({ code: "NOT_FOUND" });
+    }
+
+    eventTypeId = eventType.id;
+  }
```

Comment:
@hariombalhara I'm working on deprecating getServerSideProps entirely & also get the eventTypeId serverside.

Reply:
@alannnc can this query handle orgs event types? (we also have some extra logic in getPublicEvent function to handle this)

Reply:
@Udit-takkar I fixed so it can handle team/[slug]/[type].tsx and I see [orgSlug]/team/[slug]/[type].tsx reuses that logic. So it should work too and we have some test for that too.

Reply:
@alannnc It's not handling org event-types. I had two users in my DB with `pro` username(one in an Org so duplicate username is possible) and `getUserIdFromUsername` got me the wrong user and thus even though the booking page loaded, the schedule call gave me 404.

![image](https://github.com/calcom/cal.com/assets/1780212/c732bb16-ebc5-4040-b369-0b15ceb84e6a)

![image](https://github.com/calcom/cal.com/assets/1780212/09b0158a-1514-4406-a468-809b9228ddbc)